driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java

```diff
@@ -141,6 +144,7 @@ private Builder(final MongoClientSettings settings) {
         @SuppressWarnings("deprecation")
         public Builder applyConnectionString(final ConnectionString connectionString) {
             credentialList = new ArrayList<MongoCredential>(connectionString.getCredentialList());
+            heartbeatSocketSettingsBuilder.applyConnectionString(connectionString);
```

Comment:
I don't think it's correct to apply the connection string to the heartbeat socket settings in the same way it's applied to the regular socket settings.  Otherwise a long read timeout will become a long heartbeat read timeout, which goes against SDAM